internal/storage/clickhouse.go

```diff
@@ -346,17 +346,53 @@ func (c *ClickHouseConnector) GetTransactions(qf QueryFilter) (txs []common.Tran
 	return txs, nil
 }
 
-func (c *ClickHouseConnector) GetLogs(qf QueryFilter) (logs []common.Log, err error) {
+func (c *ClickHouseConnector) GetLogs(qf QueryFilter) (QueryResult, error) {
 	columns := "chain_id, block_number, block_hash, block_timestamp, transaction_hash, transaction_index, log_index, address, data, topic_0, topic_1, topic_2, topic_3"
-	query := fmt.Sprintf("SELECT %s FROM %s.logs FINAL WHERE is_deleted = 0%s",
-		columns, c.cfg.Database, getLimitClause(int(qf.Limit)))
-	// TODO: add aggregations, filters and grouping
+	
+	// Start building the query
+	query := fmt.Sprintf("SELECT %s FROM %s.logs FINAL WHERE is_deleted = 0", columns, c.cfg.Database)
+
+	// Add filters
+	if qf.ContractAddress != "" {
+		query += fmt.Sprintf(" AND address = '%s'", qf.ContractAddress)
```

Comment:
why not use `fmt.Sprintf` for the whole thing?
```suggestion
		query = fmt.Sprintf("%s AND address = '%s'", query, qf.ContractAddress)
```

Reply:
you mean the whole query? this way it's easier to read, we'll be adding more functionality to mane parts of the query